bigquery_etl/metadata/validate_metadata.py

```diff
@@ -27,7 +32,45 @@ def validate_public_data(metadata, path):
     return is_valid
 
 
-def validate(target):
+def validate_change_control(
+    file_path, metadata, project_id, query_files_path, codeowners_file
+):
+    """Verify that a query is correctly setup for change control."""
+    if not project_id:
+        project_id = "moz-fx-data-shared-prod"
+    if not query_files_path:
+        query_files_path = "sql"
+
+    path_to_add = file_path.partition(f"{project_id}/")[2]
+    path_in_codeowners = os.path.join(query_files_path, project_id, path_to_add)
+
+    if CHANGE_CONTROL_LABEL in metadata.labels:
+        # This label requires to have at least one owner in the metadata file.
+        # And for any of the owners, at least one entry in the CODEOWNERS file.
+        counter = 0
```

Comment:
This could also be a boolean flag `has_codeowner` that gets set if the owner is in the CODEOWNERS file. Could make the condition a little more readable